File: Easy/0283. Move Zeroes/0283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        null = None
        for index in range(len(nums)):

            if nums[index] == 0:
                if null is None:
                    null = index
                continue

            if null is None:
                continue

            nums[null] = nums[index]
            null = null + 1
            nums[index] = 0

File: Easy/0283. Move Zeroes/0283. Move Zeroes.go
func moveZeroes(nums []int)  {
    var zeros int
    last := len(nums) - 1
    for i := 0; i < last; i++ {
        if nums[i] != 0 {
            continue
        }
        zeros = 0
        for i + zeros < last && nums[i + zeros] == 0 {
            zeros++
		}
		copy(nums[i:last - zeros + 1], nums[i + zeros:last + 1])
        last -= zeros
    }
    for last = last + 1; last < len(nums); last++ {
		nums[last] = 0
    }
}
